modules/service/src/main/scala/lucuma/odb/service/DatasetService.scala

```diff
@@ -66,26 +69,28 @@ object DatasetService {
 
         import InsertDatasetResponse.*
 
-        val insert: F[Either[InsertDatasetFailure, Dataset.Id]] =
+        val insert: F[Either[InsertDatasetFailure, (Dataset.Id, Step.Id, PosShort)]] =
           session
             .unique(Statements.InsertDataset)(stepId, filename, qaState)
             .map(_.asRight[InsertDatasetFailure])
             .recover {
               case SqlState.UniqueViolation(_)     => ReusedFilename(filename).asLeft
               case SqlState.ForeignKeyViolation(_) => StepNotFound(stepId).asLeft
+              case SqlState.NotNullViolation(ex) if ex.getMessage.contains("c_observation_id") =>
+                StepNotFound(stepId).asLeft
```

Comment:
Is `c_observation_id` correct? It doesn't seem to have anything to do with the insert.

Reply:
A question is whether it's a good idea to include the observation id since it can be looked up via the step.  It's needed quite a bit though, is very convenient to have, and is a few joins away so it _seems_ like a good idea?  <grimace>

Reply:
OK. Just making sure. 😄